File: frontend/src/app/services/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { User } from '../model/User';
import { UpdateUserRequest } from '../model/dto/request/UpdateUserRequest';
import { ChangePasswordRequest } from '../model/dto/request/ChangePasswordRequest';
import { ChangeUsernameRequest } from '../model/dto/request/ChangeUsername';
import { ChangeEmailRequest } from '../model/dto/request/ChangeEmailRequest';
import * as myGlobals from 'src/globals';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  private path = myGlobals.path + '/user';

  constructor(private http: HttpClient, private router: Router) {
    this.router.routeReuseStrategy.shouldReuseRoute = () => false;
  }

  getUser() {
    return this.http.get<User>(this.path);
  }

  setUser(request: UpdateUserRequest) {
    return this.http.post<User>(this.path, request);
  }

  changePassword(data: ChangePasswordRequest) {
    return this.http.post(this.path + '/password', data);
  }

  changeUsername(request: ChangeUsernameRequest) {
    return this.http.post(this.path + '/username', request);
  }

  changeEmail(request: ChangeEmailRequest) {
    return this.http.post(this.path + '/email', request);
  }
}

File: backend/src/main/java/com/lgm/demo/service/UserService.java
package com.lgm.demo.service;

import com.lgm.demo.model.dto.request.ChangeEmailRequest;
import com.lgm.demo.model.dto.request.ChangePasswordRequest;
import com.lgm.demo.model.dto.request.UpdateUserRequest;
import com.lgm.demo.model.dto.request.ChangeUsernameRequest;
import com.lgm.demo.model.dto.response.JwtResponse;
import com.lgm.demo.model.dto.response.UserResponse;

import java.io.IOException;

public interface UserService{
    UserResponse updateUser(UpdateUserRequest request) throws IOException;
    UserResponse getUser() throws IOException;
    JwtResponse updatePassword(ChangePasswordRequest request) throws IOException;
    JwtResponse updateUsername(ChangeUsernameRequest request) throws IOException;
    JwtResponse updateEmail(ChangeEmailRequest request) throws IOException;
}
